codesign/codesign.go

```diff
@@ -0,0 +1,338 @@
+package codesign
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-xcode/appleauth"
+	"github.com/bitrise-io/go-xcode/autocodesign"
+	"github.com/bitrise-io/go-xcode/autocodesign/codesignasset"
+	"github.com/bitrise-io/go-xcode/autocodesign/devportalclient"
+	"github.com/bitrise-io/go-xcode/autocodesign/keychain"
+	"github.com/bitrise-io/go-xcode/autocodesign/projectmanager"
+	"github.com/bitrise-io/go-xcode/devportalservice"
+)
+
+// AuthType ...
+type AuthType int
+
+const (
+	// NoAuth ...
+	NoAuth AuthType = iota
+	// APIKeyAuth ...
+	APIKeyAuth
+	// AppleIDAuth ...
+	AppleIDAuth
+	// AnyAuth ...
+	AnyAuth
+)
+
+type codeSigningStrategy int
+
+const (
+	codeSigningXcode codeSigningStrategy = iota
+	codeSigningBitriseAPIKey
+	codeSigningBitriseAppleID
+)
+
+// Opts ...
+type Opts struct {
+	AuthType                   AuthType
+	ShouldConsiderXcodeSigning bool
+
+	ExportMethod      autocodesign.DistributionType
+	XcodeMajorVersion int
+
+	RegisterTestDevices bool
+	SignUITests         bool
+	MinProfileValidity  int
+	IsVerboseLog        bool
+}
+
+// Result ...
+type Result struct {
+	XcodebuildAuthParams *devportalservice.APIKeyConnection
```

Comment:
Why is it wrapped into a new struct?

Reply:
The  plan is to add other parameter that the standalone autoprov Step can use.